### ISSUE TITLE
Roads, railways, not styling correctly in QGIS

### ISSUE THREAD

**Issue by dylanparry** [OrdnanceSurvey/OS-OpenMap-Local-stylesheets#9]
Hi,

I'm using QGIS 2.18.10 and I am having problems with the styles for roads and railways in particular not being applied correctly. Here's a screenshot of what I'm seeing:

![image](https://user-images.githubusercontent.com/3692873/28458178-b1af1faa-6e00-11e7-9256-e062c95d5186.png)

As you can see, the road casings are being shown, but the fills for roads and roundabouts are not being styled correctly. I have loaded in the QML files for these layers as you should be able to see in the layers panel.

There's also a missing railway track, which appears to not be styled either. It should be running from the top left where it says "West Channel" down to the bottom right through the clearly empty channel down the in the middle of the map.

I managed to get the named places to show by following the tip in issue #2, but I see that no actual fix has been provided for this issue in terms of an updated stylesheet. The issue was closed by the reporter though. Road labels don't appear to work either.

I suspect there are other layers that aren't be styled correctly in this version of QGIS too, but I haven't seen an area that I'm familiar enough with to know what's missing yet!

Is there anything I can do to get these styles working correctly? As things are, the maps I'm seeing in QGIS are lacking so many of the styles that I cannot adequately use them for any purpose :(

Thanks,

Dylan

**Comment by egil1978** [comment]
Hi Dylan, thanks for your inquiry. Let me look into this and get back to you.

**Comment by egil1978** [comment]
Hi Dylan, 
For the layers that are not displaying would you mind checking if the scale dependent visibility box has been ticked? It can be found under the General tab within the layers properties and we have found that sometimes this box can be randomly checked for some of the layers. If it is checked then please un-check this and see if the layer draws.
Thanks

**Comment by dylanparry** [comment]
That doesn't appear to be the case here. None of the unstyled layers have that checkbox checked. I've narrowed down the problem though. It appears that the column used to categorise the styles is set to "classification" in the style, but that column doesn't exist in the ShapeFile data. There it's been truncated to "CLASSIFICA". When I select the correct column name, the styles begin to load.

There are other issues with the styling now that I have corrected the column names, as can be seen here:

![image](https://user-images.githubusercontent.com/3692873/28461941-d5b1ff8e-6e10-11e7-924b-f35a1da35645.png)

Firstly the local streets are still being drawn as solid black lines without a white fill. Then the labelling for the roads isn't quite right either: the road numbers are unreadable (what font should they be?); and the road names are also missing. Any ideas on this?

Thanks.

**Comment by egil1978** [comment]
Hi Dylan, could you tell me where you got the stylesheets from? Was it github or the download link from our website?
Many thanks.

**Comment by dylanparry** [comment]
I got them off GitHub, directly from the repository.

**Comment by uoeAndy** [comment]
I appear to have a similar problem to this although I used the OS download to get the stylesheets.  I note that the cropping to 'CLASSIFICA' described is present in both the layer attribute table and in the qml file so I don't think this is it (for me at least).

I note that unlike other data I have used (i.e. Meridian 2 - recently withdrawn) the 'road' and 'roundabout' shapefiles have no correspondingly named qml file.  Instead there are 'RoadCase.qml', RoadCentreLine.qml' and 'RoadFill.qml'.  Similarly there is 'RoundaboutCase.qml' and 'RoundaboutFill.qml'.  Applying the qml files individually seems to duplicate the results shown but I am not sure whether it is possible to apply these simultaneously or in sequence to get the required result.

**Comment by egil1978** [comment]
Hi Dylan,

Could you check you are using the stylesheets for shape files and not for GML? For GML data the attribution is classification and can cause problems with the stylesheets. You need to be loading the QML stylesheets for QGIS.
I have checked both the QML stylesheets from github and our website and the attribute for classification is classifica on both.
Many thanks

**Comment by egil1978** [comment]
Hi Andy,
You will need to duplicate your road data twice so you can apply the road centreline, road fill and road casing QML files. The layer order can be found here 
https://github.com/OrdnanceSurvey/OS-OpenMap-Local-stylesheets/blob/master/ESRI%20Shapfile%20stylesheets/QGIS%20stylesheets%20(QML)/Full%20colour%20style/images/OML_FC_layer_order.png?raw=true
You will need to duplicate your roundabout data once and apply the roundabout casing and fill QML files.
Many thanks

**Comment by dylanparry** [comment]
Hi, thanks for all this help. Now it's time for a massive facepalm as I realise I've downloaded the GML stylesheets not the ShapeFile ones! They work perfectly :)

**Comment by egil1978** [comment]
No problem Dylan. Im glad we could resolve the problem :)
Enjoy the weekend.

**Comment by uoeAndy** [comment]
Hi Paul,  I literally just arrived at this (seemingly) inevitable conclusion myself!  Thanks for your input - glad Dylan is sorted too.  ATB Andy.

**Comment by egil1978** [comment]
No problem Andy. Glad its all sorted.

**Comment by abm52** [comment]
As much as I dislike opening up old errors I have encountered something similar with QGIS3.12 and the 2019 version of OS QML files for roads and just don't seem to be able get around the problem. In a similar way I load the qml file (load style, properties dialogue and get the correct style (Road Fill) from OS loaded. The default value is 'featurecode' (in the value drop down this is FeatCode). Classification is still truncated to Classifica. 
On pressing classify this causes the OS symbology to be set up (all boxes ticked) and additional level to be added called 'all other values'. If this applied to the map a single colour symbol appears from the 'all other values' class - switch this off and no roads plot.
I have attached two images showing the initial layer properties on loading the qml file for road fill (featurecode) and a second showing Classifica option. The qml file was downloaded from OS github (https://github.com/OrdnanceSurvey/OSMM-Topography-Layer-stylesheets).
I have tried using OS Zoomstacks and they work fine with the OS style sheets.
![Classifica](https://user-images.githubusercontent.com/62698369/77643936-206c6000-6f58-11ea-9148-5e30c073ea65.jpg)
![OS_ShFi_LoadedQMLstyle](https://user-images.githubusercontent.com/62698369/77643938-2104f680-6f58-11ea-9683-14087888f3d4.jpg)
